### PR TITLE
GitHub Action fixups

### PR DESCRIPTION
**Pull request by brianhlin**
There are strange issues with missing "No such file or directory":

- https://github.com/opensciencegrid/docker-xcache/runs/1124795837?check_suite_focus=true#step:6:16
- https://github.com/opensciencegrid/docker-xcache/runs/1124795837?check_suite_focus=true#step:6:42

Since the files referenced are being created by the failing pipelines, it looks like the parent dirs aren't being created. The dirs should be created here so it's not clear why they aren't:

https://github.com/opensciencegrid/docker-xcache/runs/1124795837?check_suite_focus=true#step:6:11

My only guess was an issue with brace expansion.

### CODE REVIEW

File: .github/prep-hotfix-sources.sh [hunk 1]
```diff
@@ -1,12 +1,14 @@
...
 HOTFIX_TAG=$1
 REPO_PATH=$2
 SPEC_PATH=$3
 
 # Create OSG build dirs
 build_dir=$PWD/_build_dir
-mkdir -p "$build_dir/{osg,upstream}"
+for subdir in osg upstream; do
+    mkdir -p "$build_dir/$subdir"
+done
```
Comment (edquist) [inline .github/prep-hotfix-sources.sh:11]:
```suggestion
mkdir -p "$build_dir"/{osg,upstream}
```

I don't remember the history here, i am sorry if i misled you into this, but basically for the brace expansion to work you do not quote the braces.

File: .github/prep-hotfix-sources.sh [hunk 2]
```diff
@@ -1,12 +1,14 @@
-#!/bin/bash -x
+#!/bin/bash -xe
```
Comment (edquist) [inline .github/prep-hotfix-sources.sh:1]: yeah `set -e` is usually a good idea!

**Review by edquist — APPROVED** [review]
I'll pre approve, but do see my one suggestion

File: .github/prep-hotfix-sources.sh (hunk 1 above)
Comment (brianhlin) [reply .github/prep-hotfix-sources.sh:11]: Argh, no, that's just an oversight on my part. Good catch

**Pull request merged**